src/client/mod.rs

```diff
@@ -122,12 +126,25 @@ impl Client {
         let inner = Arc::new(ClientInner {
             topology: Topology::new(options.clone())?,
             session_pool: ServerSessionPool::new(),
+            #[cfg(feature = "fle")]
+            fle: Default::default(),
             options,
         });
-
         Ok(Self { inner })
     }
 
+    /// Creates a new `Client` connected to the cluster specified by `options` with auto-encryption
+    /// enabled.
+    #[cfg(feature = "fle")]
+    pub async fn with_fle_options(
```

Comment:
Ideally `AutoEncryptionOpts` could have just been another field in `ClientOptions`.  Unfortunately, initializing FLE state requires `async` because it has to construct a client to the mongocryptd from an arbitrary URI.  Since there needs to be a distinct creation method to accommodate that, it seemed worth keeping the options distinct to make it clear that if you're setting those options you have to use this method.